src/main/java/org/folio/marc4ld/configuration/property/Marc4BibframeRulesPostProcessor.java

```diff
@@ -0,0 +1,107 @@
+package org.folio.marc4ld.configuration.property;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Marc4BibframeRulesPostProcessor implements BeanPostProcessor {
+
+  @Override
+  public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+    if (bean instanceof Marc4BibframeRules marc4BibframeRules) {
+      postProcess(marc4BibframeRules);
+      return bean;
+    }
+    return BeanPostProcessor.super.postProcessAfterInitialization(bean, beanName);
+  }
+
+  private void postProcess(Marc4BibframeRules marc4BibframeRules) {
+    marc4BibframeRules.getFieldRules().forEach((tag, rules) ->
+      rules.forEach(rule -> {
+        var include = rule.getInclude();
+        if (include != null) {
+          copyRule(marc4BibframeRules.getSharedRules().get(include), rule);
+        }
+      })
+    );
+  }
+
+  private void copyRule(Marc4BibframeRules.FieldRule source, Marc4BibframeRules.FieldRule target) {
+    var types = source.getTypes();
+    if (types != null) {
```

Comment:
Hi @askhat-abishev,

Can we add a validation here to check if the `target.getTypes()` is empty? ie, if `target.getTypes()` already contains some value, we will retain it. We will copy value from source to target only if  that value is empty for the target.

Reply:
it looks like ```target.types``` can be either null or with some values in it
maybe it would be better to do something like this (and also add the same logic for ```subfields```, ```constants```, ```controlFields```, ```edges``` and ```mappings```)?
```java
    var sourceTypes = source.getTypes();
    if (sourceTypes != null) {
      var targetTypes = target.getTypes();
      if (targetTypes == null) {
        target.setTypes(sourceTypes);
      } else {
        targetTypes.addAll(sourceTypes);
      }
    }
```

Reply:
Yes. Good idea. Let us do it.

Reply:
done